tests/game/type-advantage.test.ts

```diff
@@ -0,0 +1,72 @@
+import {describe, expect, test} from '@jest/globals'
+import HotguyCommon from 'common/cards/alter-egos-iii/hermits/hotguy-common'
+import PoultrymanCommon from 'common/cards/alter-egos/hermits/poultryman-common'
+import {Hermit} from 'common/cards/base/types'
+import GoodTimesWithScarCommon from 'common/cards/default/hermits/goodtimeswithscar-common'
+import GrianCommon from 'common/cards/default/hermits/grian-common'
+import IJevinCommon from 'common/cards/default/hermits/ijevin-common'
+import ImpulseSVCommon from 'common/cards/default/hermits/impulsesv-common'
+import MumboJumboCommon from 'common/cards/default/hermits/mumbojumbo-common'
+import TinFoilChefCommon from 'common/cards/default/hermits/tinfoilchef-common'
+import XBCraftedCommon from 'common/cards/default/hermits/xbcrafted-common'
+import {RowComponent} from 'common/components'
+import query from 'common/components/query'
+import {attack, endTurn, playCardFromHand, testGame} from './utils'
+
+var typeToHermitMap: Map<String, Hermit> = new Map([
+	['Prankster', PoultrymanCommon],
+	['PvP', XBCraftedCommon],
+	['Redstoner', MumboJumboCommon],
+	['Speedrunner', HotguyCommon],
+	['Farm', ImpulseSVCommon],
+	['Explorer', IJevinCommon],
+	['Builder', GrianCommon],
+	['Terraformer', GoodTimesWithScarCommon],
+	['Miner', TinFoilChefCommon],
+])
```

Comment:
```suggestion
let typeToHermitMap: Map<String, Hermit> = {
	'Prankster': PoultrymanCommon,
	'PvP': XBCraftedCommon,
	'Redstoner': MumboJumboCommon,
	'Speedrunner': HotguyCommon,
	'Farm': ImpulseSVCommon,
	'Explorer': IJevinCommon,
	'Builder': GrianCommon,
	'Terraformer': GoodTimesWithScarCommon,
	'Miner': TinFoilChefCommon,
}
```

Reply:
It being var is just my bad, but typescript complains if I try to declare the map like a dictionary. The exact error is "Object literal may only specify known properties, and ''Prankster'' does not exist in type 'Map<String, Hermit>'."

Reply:
it should be `string` in lowercase I believe.

Reply:
That gives a different error when I try to get from it(using typeToHermitMap[attackingType])
`Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ Prankster: Hermit; PvP: Hermit; Redstoner: Hermit; Speedrunner: Hermit; Farm: Hermit; Explorer: Hermit; Builder: Hermit; Terraformer: Hermit; Miner: Hermit; }'.
  No index signature with a parameter of type 'string' was found on type '{ Prankster: Hermit; PvP: Hermit; Redstoner: Hermit; Speedrunner: Hermit; Farm: Hermit; Explorer: Hermit; Builder: Hermit; Terraformer: Hermit; Miner: Hermit; }'`